### DISCUSSION TITLE
Lesson 6 - Testing across networks; Error: ValueError: insufficient funds for gas * price + value

### DISCUSSION THREAD

**Discussion by ixcel87** [smartcontractkit/full-blockchain-solidity-course-py#786] [Q&A] [answered]
Hello All,

I am having an issue when running "brownie run scripts/deploy.py --network ganache-local"; error is saying "insufficient funds for gas * price + value". I get this when spinning up my local ganache instance (using ganache-cli). I suppose I am using an incorrect wallet somewhere but i cannot track it down: below is the error:

(Web3_py) ixcel87-main@ixcel87-desktop:/mnt/sdb1/2ndHome/SolidityCourse/brownie_fund_me$ brownie run scripts/deploy.py --network ganache-local
Brownie v1.17.2 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Running 'scripts/deploy.py::main'...
The active network is ganache-local
Deploying Mocks...
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/deploy.py", line 33, in main
    deploy_fund_me()
  File "./scripts/deploy.py", line 21, in deploy_fund_me
    deploy_mocks()
  File "./scripts/helpful_scripts.py", line 21, in deploy_mocks
    MockV3Aggregator.deploy(
  File "brownie/network/contract.py", line 528, in __call__
    return tx["from"].deploy(
  File "brownie/network/account.py", line 510, in deploy
    receipt, exc = self._make_transaction(
  File "brownie/network/account.py", line 752, in _make_transaction
    exc = VirtualMachineError(e)
  File "brownie/exceptions.py", line 85, in __init__
    raise ValueError(exc["message"]) from None
ValueError: insufficient funds for gas * price + value

Can someone please assist? Also, is there a way to increase the logging/debug level in brownie and ganache? 

Please let me know if you need any further information.

Total SmartContract Noob!

**Comment by ixcel87** [comment, marked as answer]
I resolved this by switching my private key in the config file to a static one from ganache dev. My meta mask key was causing this issue.

**Reply by cromewar** [reply]
Sorry I missed this question completely, I'm glad you solved the problem tough.

**Reply by ixcel87** [reply]
Thanks @cromewar

Can you please explain what this error means? I see it quite often and I believe it comes from an incorrect wallet being referenced/used, but how do I troubleshoot/debug this type of issue?

**Reply by cromewar** [reply]
Well as you know we use get_account() on our helpful scripts, and time to time the `.env` file containing the private key is not imported as it should, so the scripts will return other thing than the actual account you want.

**Reply by ixcel87** [reply]
Thanks @cromewar - another question for you please (thanks for all of your assistance), so i am trying to troubleshoot the issue even more on my own (with your guidance of course). My test_fund_me.py test is failing with this error whenever i run the following:

`(Web3_py) ixcel87-main@ixcel87-desktop:/mnt/sdb1/2ndHome/SolidityCourse/brownie_fund_me$ brownie test --network mainnet-fork-dev`

`    def test_can_fund_and_withdraw():
        account = get_account()
        deploy_fund_me()
        fund_me = FundMe[-1]
        print(f"**** The account is {account} ****")
        entrance_fee = fund_me.getEntranceFee() + 100
>       tx = fund_me.fund({"from": account, "value": entrance_fee})
E       ValueError: insufficient funds for gas * price + value

tests/test_fund_me.py:14: ValueError
------------------------------------------------------ Captured stdout call ------------------------------------------------------
Contract deployed to 0x7b4BbE50e63B819AFA670f0C80A314B1dA4E5995
**** The account is 0x27F740bf04FE181463c4a3C763B3544B874569f5 ****
==================================================== short test summary info =====================================================
FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - ValueError: insufficient funds for gas * price + value
================================================== 1 failed, 1 skipped in 3.93s ==================================================
Terminating local RPC client...`

I used your advice and printed out the details of the account with the below statement:
`print(f"**** The account is {account} ****")`
but how do i print out EVERYTHING about the account? Is there a simple way to get the details of the account?

Thanks in advance!

**Reply by cromewar** [reply]
Is  mainnet-fork-dev using your metamask wallet? you would need to have more funds on it in order to work.

Also you can check a lot more information about brownie accounts here:  https://eth-brownie.readthedocs.io/en/stable/core-accounts.html

**Reply by ixcel87** [reply]
Hey @cromewar 

I do not believe it is using my metamask wallet. I have 2 wallets with metamask with approximately 1.7 ETH and 0.9 ETH, both on the Rinkeby test network. (accts: 0x58D02301616baaf3e4d6aD746392674aE97d32Bf and 0xe7E322403D6d4407B021C3EF647fE72Cd29cAAE6 - feel free to send me some ETH!)

A couple of things i noticed:
* if i spin up a ganache-cli command using forked network using `ganache-cli --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/dUPrhMlUbymDaQ7vDQZFN3ba9ibOnyiX --mnemonic brownie --port 8545 --hardfork istanbul`; i get the following accounts: 
![image](https://user-images.githubusercontent.com/41350932/150690405-0b0765de-f3bd-49df-a060-47865e94a785.png)

* when i run `(Web3_py) ixcel87-main@ixcel87-desktop:/mnt/sdb1/2ndHome/SolidityCourse/brownie_fund_me$ brownie test --network mainnet-fork-dev`; my output for account is:
![image](https://user-images.githubusercontent.com/41350932/150690482-3dcd4c24-3c8f-435f-8cd8-dae3c81dc255.png)

Where is this 0x27F740bf04FE181463c4a3C763B3544B874569f5 account coming from?

* i do see the contract deployed @ 0x7b4BbE50e63B819AFA670f0C80A314B1dA4E5995 listed in my 7545 build - could this be the smoking gun?
![image](https://user-images.githubusercontent.com/41350932/150690586-c36e9332-c400-411d-98a5-134033ffabd0.png)

**Reply by cromewar** [reply]
Each account will change each time you lunch ganache unless you use the --deterministic which will give you always the same account addresses, refer [here](https://docs.nethereum.com/en/latest/ethereum-and-clients/ganache-cli/) for more info, and try using it to see what's going on.
